File: app/src/main/java/net/north101/android/ghplayertracker/ClassViewHolder.java
package net.north101.android.ghplayertracker;

import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

public class ClassViewHolder extends BaseViewHolder<String> {
    public static int layout = R.layout.class_item_view;

    private final ImageView itemView;

    public static ClassViewHolder inflate(ViewGroup parent) {
        return new ClassViewHolder(inflate(parent, layout));
    }

    public ClassViewHolder(View view) {
        super(view);

        this.itemView = view.findViewById(R.id.class_icon);
    }

    public void bind(String item) {
        super.bind(item);

        this.itemView.setImageResource(Util.getImageResource(itemView.getContext(), item));
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/ClassViewHolder.kt
package net.north101.android.ghplayertracker

import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import net.north101.android.ghplayertracker.data.CharacterClass

class ClassViewHolder(view: View) : BaseViewHolder<CharacterClass>(view) {
    private val imageView: ImageView = view.findViewById(R.id.class_icon)

    override fun bind(item: CharacterClass) {
        super.bind(item)

        this.imageView.setImageResource(Util.getImageResource(imageView.context, item.id))
    }

    companion object {
        var layout = R.layout.class_item_view

        fun inflate(parent: ViewGroup): ClassViewHolder {
            return ClassViewHolder(BaseViewHolder.inflate(parent, layout))
        }
    }
}
